webapi/src/main/scala/org/knora/webapi/routing/admin/ProjectsRouteZ.scala

```diff
@@ -85,18 +91,32 @@ final case class ProjectsRouteZ(
   private def deleteProject(iriUrlEncoded: String, requestingUser: UserADM): Task[Response] =
     for {
       iriDecoded            <- RouteUtilZ.urlDecode(iriUrlEncoded, s"Failed to URL decode IRI parameter $iriUrlEncoded.")
-      projectIri            <- Iri.ProjectIri.make(iriDecoded).toZIO.mapError(e => BadRequestException(e.msg))
+      projectIri            <- ProjectIri.make(iriDecoded).toZIO.mapError(e => BadRequestException(e.msg))
       projectDeleteResponse <- projectsService.deleteProject(projectIri, requestingUser)
     } yield Response.json(projectDeleteResponse.toJsValue.toString())
 
   private def updateProject(iriUrlEncoded: String, request: Request, requestingUser: UserADM): Task[Response] =
     for {
       iriDecoded            <- RouteUtilZ.urlDecode(iriUrlEncoded, s"Failed to URL decode IRI parameter $iriUrlEncoded.")
-      projectIri            <- Iri.ProjectIri.make(iriDecoded).toZIO.mapError(e => BadRequestException(e.msg))
+      projectIri            <- ProjectIri.make(iriDecoded).toZIO.mapError(e => BadRequestException(e.msg))
       body                  <- request.body.asString
       payload               <- ZIO.fromEither(body.fromJson[ProjectUpdatePayloadADM]).mapError(e => BadRequestException(e))
       projectChangeResponse <- projectsService.updateProject(projectIri, payload, requestingUser)
     } yield Response.json(projectChangeResponse.toJsValue.toString)
+
+  private def getAllProjectData(iriUrlEncoded: String, requestingUser: UserADM): Task[Response] =
+    for {
+      iriDecoded             <- RouteUtilZ.urlDecode(iriUrlEncoded, s"Failed to URL decode IRI parameter $iriUrlEncoded.")
+      iriIdentifier          <- IriIdentifier.fromString(iriDecoded).toZIO.mapError(e => BadRequestException(e.msg))
+      projectDataGetResponse <- projectsService.getAllProjectData(iriIdentifier, requestingUser)
+      filePath                = projectDataGetResponse.projectDataFile
+      response = Response(
+                   headers = Headers.contentType("application/trig"),
+                   body = Body.fromFile(new File(filePath.toString()))
+                 )
+      _ = ZIO.succeed(ZIO.attempt(Files.deleteIfExists(filePath)))
```

Comment:
would be nice to have some error logging in case the attempt is failing.

Reply:
done